receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -351,483 +322,205 @@ func (c *kafkaLogsConsumer) Start(_ context.Context, host component.Host) error
 		return err
 	}
 
-	unmarshaler, err := newLogsUnmarshaler(c.config.Logs.Encoding, c.settings, host)
+	consumerGroup, err := kafka.NewSaramaConsumerGroup(
+		context.Background(),
+		c.config.ClientConfig,
+		c.config.ConsumerConfig,
+	)
 	if err != nil {
 		return err
 	}
-	c.unmarshaler = unmarshaler
+	c.consumerGroup = consumerGroup
 
-	// consumerGroup may be set in tests to inject fake implementation.
-	if c.consumerGroup == nil {
-		if c.consumerGroup, err = createKafkaClient(ctx, c.config); err != nil {
-			return err
-		}
-	}
-	logsConsumerGroup := &logsConsumerGroupHandler{
+	handler := &consumerGroupHandler{
 		logger:            c.settings.Logger,
-		encoding:          c.config.Logs.Encoding,
-		unmarshaler:       c.unmarshaler,
-		nextConsumer:      c.nextConsumer,
 		ready:             make(chan bool),
 		obsrecv:           obsrecv,
-		autocommitEnabled: c.autocommitEnabled,
-		messageMarking:    c.messageMarking,
-		headerExtractor:   &nopHeaderExtractor{},
+		autocommitEnabled: c.config.AutoCommit.Enable,
+		messageMarking:    c.config.MessageMarking,
 		telemetryBuilder:  c.telemetryBuilder,
 		backOff:           newExponentialBackOff(c.config.ErrorBackOff),
 	}
-	if c.headerExtraction {
-		logsConsumerGroup.headerExtractor = &headerExtractor{
-			logger:  c.settings.Logger,
-			headers: c.headers,
-		}
+	consumeMessage, err := c.newConsumeMessageFunc(handler, host)
+	if err != nil {
+		return err
 	}
-	c.consumeLoopWG.Add(1)
-	go c.consumeLoop(ctx, logsConsumerGroup)
-	<-logsConsumerGroup.ready
+	handler.consumeMessage = consumeMessage
+
+	c.consumeLoopClosed = make(chan struct{})
+	c.started = true
+	go c.consumeLoop(handler)
```

Comment:
Not within this PR, but for an upcoming change, could these be moved to `errgroup.Group`?

This is mostly to simplify the logic around ensuring spawned routines are correctly processed and shutdown on component shutdown.

Reply:
I think it could, but we might lose the benefit of using the consumeLoopClosed channel, which is that waiting for it can be interrupted

Reply:
Resolving, but again happy to discuss suggestions